### PR TITLE
Initial teamcity output support.

### PR DESCRIPTION
**Pull request by neochief**
See https://github.com/paratestphp/paratest/issues/595

### CODE REVIEW

File: src/Runners/PHPUnit/Options.php [hunk 1]
```diff
@@ -1078,14 +1090,19 @@ public function logJunit(): ?string
...
+
     public function logTeamcity(): ?string
     {
         return $this->logTeamcity;
     }
 
     public function hasLogTeamcity(): bool
     {
-        return $this->logTeamcity !== null;
+        return $this->teamcity || $this->logTeamcity !== null;
```
Comment (Slamdunk) [inline src/Runners/PHPUnit/Options.php:1105]: Output and logging are two different things and shouldn't be coupled

File: src/Runners/PHPUnit/ResultPrinter.php [hunk 2]
```diff
@@ -122,6 +126,9 @@ public function __construct(LogInterpreter $results, OutputInterface $output, Op
         $this->output  = $output;
         $this->options = $options;
 
+        $this->printsTeamcity = $this->options->teamcity();
+        $this->needsTeamcity = $this->printsTeamcity || $this->options->logTeamcity();
```
Comment (Slamdunk) [inline src/Runners/PHPUnit/ResultPrinter.php:130]: Same here: output and logging are two different things and shouldn't be coupled

**Review by Slamdunk — CHANGES_REQUESTED** [review]
Before reviewing the code, I'd like to review the tests, which are missing, please write the tests before the code

File: src/Runners/PHPUnit/ResultPrinter.php (hunk 2 above)
Comment (neochief) [reply src/Runners/PHPUnit/ResultPrinter.php]: I have moved this check into a method in the Options: https://github.com/paratestphp/paratest/pull/597/files#diff-e7eb5918ab64b6a45aac1cb6586bd195c5a96fd600882a7be2478afe58edbe5cR1108

**Comment by neochief** [comment]
> Before reviewing the code, I'd like to review the tests, which are missing, please write the tests before the code

Okay, I've added a few tests and assertions.

File: src/Runners/PHPUnit/Options.php (hunk 1 above)
Comment (Slamdunk) [reply src/Runners/PHPUnit/Options.php]: This point doesn't seem resolved to me: a user could use `--teamcity` but not `--log-teamcity` and viceversa, please split the two behaviors

Comment (neochief) [reply src/Runners/PHPUnit/Options.php]: Hm, this seems to be already changed, see:
https://github.com/paratestphp/paratest/pull/597/files#diff-e7eb5918ab64b6a45aac1cb6586bd195c5a96fd600882a7be2478afe58edbe5cR1107

The resulting code looks like this:
```
...
public function hasLogTeamcity(): bool
{
    return $this->logTeamcity !== null;
}

public function needsTeamcity(): bool
{
    return $this->teamcity() || $this->hasLogTeamcity();
}
...
```

**Comment by neochief** [comment]
PR can't pass static analysis checks, but it seems that the remaining issues are also present in master.

**Comment by pyrou** [comment]
Any update on this?

**Comment by Slamdunk** [comment]
See https://github.com/paratestphp/paratest/issues/595#issuecomment-799505182

I thank you @neochief for the effort spent in this PR, but this approach doesn't fit the goal; we need a separate binary to serve Paratest to PHPStorm disguised as PHPUnit.

I'm busy right now: any help would be welcome; I'm going to keep this open only as a reminder

**Comment by Slamdunk** [comment]
It was not my intention to close this PR while renamed the default branch.

Nevertheless my last comment is still valid, hope to see someone implementing it

**Comment by Slamdunk** [comment]
Thank you @neochief , this PR has been integrated in https://github.com/paratestphp/paratest/pull/672 and merged

**Pull request closed without merging**